actioncable/lib/action_cable/subscription_adapter/postgresql.rb

```diff
@@ -123,9 +131,16 @@ def remove_channel(channel)
             @queue.push([:unlisten, channel])
           end
 
-          def invoke_callback(*)
+          def invoke_callback(callback, message)
+            message = decoder.decode(message)
+            message = Zlib.inflate(message) if message.start_with?("\x78")
```

Comment:
Let's better use custom prefixes to distinguish compressed and uncompressed  payloads. It could be as simple as "0#{uncompressed}" and "1#{compressed}" (so, just a compression bit).

That would make it possible to plug custom compressors in the future.